base/iterators.jl

```diff
@@ -882,6 +890,7 @@ function iterate(it::Cycle, state)
 end
 
 reverse(it::Cycle) = Cycle(reverse(it.xs))
+last(it::Cycle) = last(it.xs)
```

Comment:
This (and the method on the previous line!) doesn't seem right. The sequence doesn't converge.